src/commands/reword.rs

```diff
@@ -63,11 +65,12 @@ pub fn reword(
     };
 
     let edit_message_fn = |message: &str| {
-        // Editor::edit() normally requires that the file being edited is saved before the editor
-        // is closed. If it's not, it will return None; and in all other cases, it will return
-        // Some(message). We don't care if the file has been saved, only if it hasn't changed, so
-        // here we call `require_save(false)` and just ignore the None case.
-        let result = Editor::new()
+        let mut editor = Editor::new();
+        let editor = match get_editor(&repo)? {
+            Some(prog) => editor.executable(prog),
+            None => &mut editor,
+        };
+        let result = editor
```

Comment:
This works just fine, but I'm 100% open to suggestions if there's a more idiomatic way to handle optional chaining calls in Rust. The only other thing that came to my mind was to have `get_editor` do all of the resolution, and then just always call `.executable(get_editor(&repo)?.unwrap())` or similar here. Doing all of the resolution would probably be fine, but it felt like overkill since dialoguer already does some of it, and my implementation would just be copying from them, I think. 😄

Reply:
This seems good to me. Let's not duplicate the logic inside `Editor` itself. I also notice that builder patterns tend to be a little clunky in Rust.